bundles/cif-connector-graphql/src/main/java/com/adobe/cq/commerce/gui/components/common/cifproductfield/impl/SearchDataSourceServlet.java

```diff
@@ -12,63 +12,77 @@
  *
  ******************************************************************************/
 
-package libs.commerce.gui.components.common.cifproductfield.datasources.search;
+package com.adobe.cq.commerce.gui.components.common.cifproductfield.impl;
 
 import java.io.IOException;
-
-import java.util.Map;
 import java.util.HashMap;
 import java.util.Iterator;
+import java.util.Map;
 
-import javax.servlet.ServletException;
+import javax.servlet.Servlet;
 
-import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.apache.commons.collections4.iterators.TransformIterator;
 import org.apache.sling.api.SlingHttpServletRequest;
 import org.apache.sling.api.SlingHttpServletResponse;
-
-import org.apache.commons.collections4.Transformer;
-import org.apache.commons.collections4.iterators.TransformIterator;
 import org.apache.sling.api.resource.Resource;
 import org.apache.sling.api.resource.ResourceWrapper;
 import org.apache.sling.api.scripting.SlingBindings;
 import org.apache.sling.api.scripting.SlingScriptHelper;
+import org.apache.sling.api.servlets.SlingSafeMethodsServlet;
+import org.osgi.service.component.annotations.Component;
+
+import com.adobe.cq.commerce.graphql.search.CatalogSearchSupport;
 import com.adobe.granite.ui.components.Config;
 import com.adobe.granite.ui.components.ExpressionHelper;
 import com.adobe.granite.ui.components.ExpressionResolver;
 import com.adobe.granite.ui.components.ds.DataSource;
 import com.adobe.granite.ui.components.ds.SimpleDataSource;
 import com.fasterxml.jackson.databind.ObjectMapper;
 
-public class search extends SlingSafeMethodsServlet {
-    protected void doGet(SlingHttpServletRequest request, SlingHttpServletResponse response)
-            throws ServletException, IOException {
-        final String VIRTUAL_PRODUCT_QUERY_LANGUAGE = "virtualProductOmnisearchQuery";
+import static com.adobe.cq.commerce.graphql.resource.GraphqlQueryLanguageProvider.*;
+
+@Component(
+    service = Servlet.class,
+    name = "CIFProductFieldSearchDataSourceServlet",
```

Comment:
Don't we want to have the same name as the class itself?

Reply:
I preferred a more descriptive name for the component here than the class name. I think this is only relevant for the human reader in the OSGi console and it defaults to the fully qualified Java class name if it's not specified explicitly.